tests/internal/test_module.py

```diff
@@ -502,3 +502,14 @@ def test_module_watchdog_importlib_resources_files():
     import importlib.resources as r
 
     assert isinstance(r.files("namespace_test"), MultiplexedPath)
+
+
+@pytest.mark.subprocess
+def test_module_watchdog_does_not_rewrap():
+    """Ensures that self.loader.get_code() does not get re-wrapped (leading to a deeper stack)"""
+    from importlib import reload
+
+    from ddtrace import internal
+
+    for _ in range(10000):
```

Comment:
How long does this test take? I think it's valuable to have the repro condition as a test, but just curious what the duration is for this loop

Reply:
In its new iteration:
```
0.54s call     tests/internal/test_module.py::test_module_watchdog_reloads_dont_cause_errors
```
Prior to that it was on the order of 3s or so.